### PR TITLE
TYP, DOC, CLN:SeriesGroupBy._wrap_applied_output

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -413,7 +413,25 @@ def _wrap_transformed_output(
         assert isinstance(result, Series)
         return result
 
-    def _wrap_applied_output(self, keys, values, not_indexed_same=False):
+    def _wrap_applied_output(
+        self, keys: Index, values: List[Any], not_indexed_same: bool = False
```
Comment (WillAyd) [inline pandas/core/groupby/generic.py:417]: Are values actually `List[Any]` here? Thought we passed ndarrays as well

**Review by WillAyd — COMMENTED** [review]
Minor comment but I think generally OK. There might be some overlap here with what @TomAugspurger is working on

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/core/groupby/generic.py]: @WillAyd I added
```python
assert isinstance(values, list) or values is None
```
and ran
```bash
$ pytest pandas/tests/groupby/ --pdb
```
, which passed - is there anywhere else that would pass an `ndarray`?

(this comment got me to correct it from `List` to `Optional[List]`, so thanks for that!)

Comment (WillAyd) [reply pandas/core/groupby/generic.py]: Sounds like I was wrong on that then :-) thanks for checking

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by TomAugspurger** [comment]
Thanks @MarcoGorelli.

**Pull request merged**